serverlessworkflow/modules/ROOT/pages/core/understanding-workflow-error-handling.adoc

```diff
@@ -1 +1,77 @@
+
+The algorithm used by Kogito to determine if an exception should be handled by an error definition is:
+
+* It checks if the type of the Java Exception being thrown is an instance (which includes any descendant) of the Java Exception specified in `code` property of the error definition. 
+* If this not the case, it recursively checks if the root cause of the Java Exception is an instance (which includes any descendant) of the Java Exception specified in `code` property of the error definition.
+
+== Error handling example 
+
+Lets illustrate error handling with this link:{kogito_sw_examples_url}/serverless-workflow-error-quarkus[Serverless Workflow Application]. 
+
+image::core/error_handling.png []
+
+This workflow accepts an integer number as input model. When the workflow starts, it invokes `isEven` service passing that number as parameter. This link:{kogito_sw_examples_url}serverless-workflow-error-quarkus/src/main/java/org/kie/kogito/examples/EvenService.java[Java service] throws a `IllegalArgumentException` exception if the number is odd. This means that if the number is even, the flow will transition to the `even` state, but if the number is odd, flow will move to the `odd` state, resulting in a different model output. The link:{spec_doc_url}#inject-state[Inject] state purpose is to populate the model with some specific JSON payload, therefore the `even` state sets the `message` property to __even__ and the `odd` state consequently sets it to __odd__. The `finish` state dumps the model content to the console, so user can verify the expected message has been set. 
+
+This functionality is achieved by:
+
+* Defining a function that invokes the `isEven` method of `org.kie.kogito.examples.EvenService` class. This function use link:{spec_doc_url}#defining-custom-function-types[custom type] to enable the workflow to invoke java methods.
++
+[source, json]
+----
+{
+    "name": "isEven",
+    "type": "custom",
+    "operation": "service:java:org.kie.kogito.examples.EvenService::isEven"
+}
+----
+* Adding an error definition that handles any runtime exception at the beginning of the link:{kogito_sw_examples_url}/serverless-workflow-error-quarkus/src/main/resources/error.sw.json[workflow]. 
++
+[source, json]
+----
+"errors": [
+    {
+      "name": "odd number",
+      "code": "java.lang.RuntimeException"
+     }
+  ]
+----
+* Referring this error definition in the 'checkEven' state to tell the workflow that if this error occurs it should transition to `odd` state rather than the regular transition to `even` state. Since `IllegalArgumentException` is a child of `RuntimeException`, it will be handled by the `errorRef` if thrown by `isEven` function.
```

Comment:
```suggestion
* Referring to this error definition in the 'checkEven' state to tell the workflow that if this error occurs, it should transition to an `odd` state rather than the regular transition to the `even` state. Since `IllegalArgumentException` is a child of `RuntimeException`, it will be handled by the `errorRef` if thrown by the `isEven` function.
```